### PR TITLE
Martin/coordinator updates

### PR DESCRIPTION
**Pull request by MartinJibber**

### CODE REVIEW

File: Shared/Coordinator/Coordinator.swift [hunk 1]
```diff
@@ -28,33 +40,37 @@ class Coordinator<Result>: NSObject, CoordinatorType {
...
     func addChildAndStart<ChildResult>(_ coordinator: Coordinator<ChildResult>,
                                        finishedHandler: @escaping (ChildResult) -> Void) {
-        guard self.childCoordinator == nil else {
-            print("WARNING!!!!! ATTEMPTING TO ADD CHILD COORDINATOR \(coordinator)"
-                + " TO COORDINATOR \(self) THAT ALREADY HAS ONE \(self.childCoordinator!)")
-            return
+        // If we already have a child coordinator, log a warning. While this isn't ideal, it helps
+        // prevent apps from getting locked up due to a coordinator not finishing or being presented
+        // properly.
+        if self.childCoordinator != nil {
```
Comment (Benjamindodgson) [inline Shared/Coordinator/Coordinator.swift]: You can use .nil

File: Shared/Coordinator/Router.swift [hunk 2]
```diff
@@ -93,40 +113,80 @@ class Router: NSObject, UINavigationControllerDelegate {
         self.navController.pushViewController(viewController, animated: animated)
     }
 
-    /// Pops the top module, can pass in a module if the one you need to pop is not on the top of the stack, but you must ensure that the instance will be the same as the one in the navigation stack (aka to presentable does not create a new instance)
-    func popModule(animated: Bool, module: Presentable? = nil) {
+    /// Similar to a push, however the current view controller will removed from the stack once the new one is pushed, effectively making it a swap.
```
Comment (Benjamindodgson) [inline Shared/Coordinator/Router.swift]: * will be

**Review by Benjamindodgson — CHANGES_REQUESTED** [review]
2 minor things.

File: Shared/Coordinator/Router.swift (hunk 2 above)
Comment (MartinJibber) [reply Shared/Coordinator/Router.swift]: Haha nice catch! This function was written by Ethan technically so it's his original typo.

**Pull request merged**